### ISSUE TITLE
doc: cmd/go: some module docs are on swtch.com, not golang.org

### ISSUE THREAD

**Issue by wsc1** [golang/go#27220]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?

golang.org today
### Does this issue reproduce with the latest release?

yes
### What operating system and processor architecture are you using (`go env`)?

I don't think it matters

### What did you do?

went to golang.org
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
A link on play.golang.org is best.

### What did you expect to see?

Documentation for the go programming language to be hosted on golang.org

### What did you see instead?

Documentation for go modules hosted on swtch.com

**Comment by ALTree** [comment]
It's not clear what is that it's only available on swtch.com and you'd like to see on golang.org. Can you expand?

**Comment by wsc1** [comment]
Yes, but I cannot elaborate it all without going through everything.

There are some links I came across,  mostly in the https://golang.org/cmd/go/ page
which refer to swtch.com for information not present on golang.org.

To my taste, I would prefer golang.org docs be self contained.

With full understanding it's not up to me;  just would like to express the opinion that
the implications of pointing to a .com from a .org for more info don't sit well, and would ideally be self-contained.  nothing against the content of swtch.com intended.

**Comment by ianlancetaylor** [comment]
It's not reasonable for us to refuse to link to any other site from golang.org.  For example, https://golang.org/pkg/html/template/ links to rawgit.com.  https://golang.org/pkg/archive/zip/ links to pkware.com.  I'm sure there are many other examples.

That said, yes, the vgo information on swtch.com is Go-specific and could reasonably move to golang.org.  Modules are currently an experimental facility.  Let's revisit this when we are closer to the 1.12 release.

**Comment by wsc1** [comment]
Yes of course I meant go specific documentation, but should have specified.

Last I checked, the detailed assembler docs were old plan9 and I think off-site, but now it appears correctly referenced as background at 9p.io.

**Comment by agnivade** [comment]
There are currently 3 links to https://research.swtch.com/ in the cmd/go page.

- https://research.swtch.com/vgo-mvs is in https://golang.org/cmd/go/#hdr-The_go_mod_file.
- https://research.swtch.com/vgo-import is in https://golang.org/cmd/go/#hdr-Module_compatibility_and_semantic_versioning
And
- https://research.swtch.com/vgo-module is in https://golang.org/cmd/go/#hdr-Module_code_layout.

Only [Modular code layout](https://golang.org/cmd/go/#hdr-Module_code_layout) leans heavily on research.swtch.com saying - 

> For now, see https://research.swtch.com/vgo-module for information about how source code in version control systems is mapped to module file trees.

All others just have supplemental material ponting to research.swtch.com.

@rsc - Do you have plans to move some of the modular code layout section inside golang.org ?

**Comment by matloob** [comment]
I don't think this is an issue anymore? Searching the go command package documentation there aren't any research.swtch.com links.
